code/web/src/modules/user/api/state.test.js

```diff
@@ -0,0 +1,49 @@
+import userReducer from './state.js'
+
+describe('userReducer', () => {
+  it('should return initial state', () => {
+    // Setup
+    const expected = {
+      "details": null,
+      "error": null,
+      "isAuthenticated": false,
+      "isLoading": false,
+    }
+
+    // Execution
+    const result = userReducer(undefined, {});
+
+    // Expectation
+    expect(result).toEqual(expected);
+  });
+
+  it('should return the updated state', () => {
+    // Setup
+    const initialState = {
+      "details": null,
+      "error": null,
+      "isAuthenticated": false,
+      "isLoading": false,
+    }
+
+    const user = { name: "Bob" }
+
+    const action = {
+      type: "AUTH/GET_USER_DETAILS", 
```

Comment:
Awesome first two tests! I think we could add one more here for the editDetails method and then move on to testing the action creators.